Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -79,7 +79,75 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
                 }
             }
 
-            return new PlanarSurface { ExternalBoundary = externalBoundary, InternalBoundaries = internalBoundaries };
+            //// Internal checks independent from External
+            // Selfintersections Internal
+            count = internalBoundaries.Count;
+
+            internalBoundaries = internalBoundaries.Where(x => x is NurbsCurve || !x.IIsSelfIntersecting()).ToList();
+
+            if (internalBoundaries.Count != count)
+            {
+                Reflection.Compute.RecordWarning("At least one of the internalBoundaries is selfintersecting. And have been disregarded.");
+            }
+
+            // Runs Boolean Union to resolve intersections and containemnt
+            count = internalBoundaries.Count;
+
+            internalBoundaries = internalBoundaries.Where(x => x is NurbsCurve || x is Ellipse)
+                                .Concat(internalBoundaries.Where(x => !(x is NurbsCurve) && !(x is Ellipse)).BooleanUnion()).ToList();
+
+            if (internalBoundaries.Count != count)
+            {
+                Reflection.Compute.RecordWarning("At least one of the internalBoundaries was overlapping another one. BooleanUnion was used to resolve it.");
+            }
+
+            //// Checks based around external curve
+            if (externalBoundary is NurbsCurve)
+            {
+                Reflection.Compute.RecordWarning("Neeseccary checks to ensure vadility of the PlanarSurface is not implemented. The PlanarSurfaces curves relations are not garanteueed.");
+                // External done
+                return new PlanarSurface(externalBoundary, internalBoundaries);
+            }
+            else if (externalBoundary.IIsSelfIntersecting())
+            {
+                Reflection.Compute.RecordError("The provided externalBoundary is selfintersecting.");
+                return null;
+            }
+            if (externalBoundary is Ellipse)
+                return new oM.Geometry.PlanarSurface(externalBoundary, internalBoundaries);
+
+
+            // Intersects External
+            for (int i = 0; i < internalBoundaries.Count; i++)
+            {
+                ICurve intCurve = internalBoundaries[i];
+                if (intCurve is NurbsCurve || intCurve is Ellipse)
```

Comment:
Please check (somewhere close to the beginning) if there are any internal curves that are ellipses or nurbs and record an error that they will be not checked?